relay_internal_test.go

```diff
@@ -46,3 +47,35 @@ func TestFinishesCallResponses(t *testing.T) {
 		assert.Equal(t, tt.finishesCall, finishesCall(f), "Wrong isLast for flags %v and message type %v", tt.flags, tt.msgType)
 	}
 }
+
+func TestRelayTimerPoolMisuse(t *testing.T) {
+	tests := []struct {
+		msg string
+		f   func(*relayTimer)
+	}{
+		{
+			msg: "release without stop",
+			f: func(rt *relayTimer) {
+				rt.Start(time.Hour, &relayItems{}, 0, false /* isOriginator */)
+				rt.Release()
+			},
+		},
+		{
+			msg: "start twice",
+			f: func(rt *relayTimer) {
+				rt.Start(time.Hour, &relayItems{}, 0, false /* isOriginator */)
+				rt.Start(time.Hour, &relayItems{}, 0, false /* isOriginator */)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		trigger := func(*relayItems, uint32, bool) {}
+		rtp := newRelayTimerPool(trigger)
+
+		rt := rtp.Get()
+		assert.Panics(t, func() {
+			tt.f(rt)
+		}, tt.msg)
+	}
+}
```

Comment:
Can we add a unit test (vs the integration tests elsewhere) that ensures that `Stop` is safe to use concurrently?

Reply:
Yep, added a test which calls `Stop` and tries to race it against itself, but also against the timer firing.